src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java

```diff
@@ -215,6 +269,44 @@ public void testRecordConsumption() throws Exception {
                 }).get();
     }
 
+    @Test
+    public void testRecordDeleteFieldWithYBExtractNewRecordState() throws Exception {
+        TestHelper.dropAllSchemas();
+        TestHelper.executeDDL("yugabyte_create_tables.ddl");
+        Thread.sleep(1000);
+
+        YBExtractNewRecordState<SourceRecord> transformation = new YBExtractNewRecordState<>();
+
+        Map<String, Object> configs = new HashMap<String, Object>();
+        configs.put(ExtractNewRecordStateConfigDefinition.DROP_TOMBSTONES.toString(), "false");
+        configs.put(ExtractNewRecordStateConfigDefinition.HANDLE_DELETES.toString(), "rewrite");
```

Comment:
Can we add test for different delete modes

Reply:
The other two modes of "HANDLE_DELETE" are "drop" and "none"
And in both cases, the "__deleted" field won't be populated

Reply:
Yes, some verification that the record is as expected.